[docs] Redesign on markdown page

mnajdova opened the pull request:
Changes done:

## Table content:
Prev:

<img src="https://user-images.githubusercontent.com/4512430/130226464-5cb9406a-4b72-4b2a-91c1-02d32108eca2.png" width="150px" />

Now:

<img src="https://user-images.githubusercontent.com/4512430/130226446-c5b01b28-c1c7-4ea1-b78e-689383890f47.png" width="150px" />

## Demo toolbar

Prev:

<img src="https://user-images.githubusercontent.com/4512430/130226675-4b8e8383-0da9-4be9-ac81-a2bc2e640faf.png" width="200px" />

Now:

<img src="https://user-images.githubusercontent.com/4512430/130226744-d6a55e34-b040-4b66-b577-25dc8fa16a7d.png" width="200px" />

## Page heading

Prev:

<img src="https://user-images.githubusercontent.com/4512430/130227022-42b571a1-a24f-4139-b0f4-f9473ec368be.png" widith="300px" />

Now:

<img src="https://user-images.githubusercontent.com/4512430/130226985-ed34868f-fb14-421e-b307-9153aa8393d2.png" widith="300px" />

TODO:
## API link

Prev:

<img src="https://user-images.githubusercontent.com/4512430/130234465-0fc12112-04cb-498b-bbc3-2bd4d7ad088e.png" width="150px" />


Now:

<img src="https://user-images.githubusercontent.com/4512430/130234408-b5782b21-d229-4fe0-8b04-bdabdd5a650c.png" width="150px" />

## Updated the colors on the content of the docs pages

Next iteration:
- Sidebar

---

Scrollbar dark mode - will be handled after https://github.com/mui-org/material-ui/issues/25016 is fixed

mnajdova commented:
I noticed one problem when I played with the scrollbar, namely the global styles ar propagated to the demos too (see the scrollbar on the demo) Not sure what we should do with this one. cc @siriwatknp @oliviertassinari 

![image](https://user-images.githubusercontent.com/4512430/130230840-40446a22-1022-494f-ba7c-16f581c1064a.png)


I feel like in the end we may need to isolate all examples in iframes :\

mnajdova commented on docs/src/modules/components/MarkdownElement.js:
```diff
@@ -31,17 +33,14 @@ const Root = styled('div')(({ theme }) => ({
   },
   '& code[class*="language-"]': {
-    backgroundColor: '#272c34',
+    backgroundColor: "#001E3C",
```
We should check why sometimes using `theme.palette.primaryDark[700]` is throwing. cc @siriwatknp

siriwatknp replied:
I might have missed it.

danilo-leal commented:
Nice work! I've pushed some small changes here and there. Listing some other ones I couldn't do:

1. The texts and the demo/code containers are slightly misaligned horizontally. Couldn't find where to play with this...
<img width="500" alt="Screen Shot 2021-08-20 at 12 44 36" src="https://user-images.githubusercontent.com/67129314/130260418-af0dcd93-7794-481d-a12d-b4963a874bbc.png">


2. There are some cases where the demo and code container don't have border-radius applied. No idea why...
<img width="500" alt="Screen Shot 2021-08-20 at 12 39 17" src="https://user-images.githubusercontent.com/67129314/130260464-4d458c83-02f3-4714-a3b8-807cdf614580.png">

3. Couldn't find where to modify the colors of the filled and outlined demo container on dark mode. 
<img width="500" alt="Screen Shot 2021-08-20 at 12 56 42" src="https://user-images.githubusercontent.com/67129314/130260644-41f98c03-9960-4927-8cda-7bbc71a7e2ff.png">

4. Couldn't find where to modify the max-width of the container that holds all the content but I played with some values directly on the inspector. This is something I personally think it makes for a much better reading experience (less horizontal movement of the eye) and also for a more overall balanced look since we gain negative space between the sidebar and the table of contents. 
<img width="600" alt="Screen Shot 2021-08-20 at 12 58 44" src="https://user-images.githubusercontent.com/67129314/130260957-2ec1cadc-0872-4387-a72e-d70719786c17.png">
These values are just me playing, not necessarily the correct ones. Probably somewhere around 700px - 800px lies a good max-width. For reference, most of our users check the website in a 1920x1080 screen res which renders the div with 1152 width, which is too much 😅 But we can iterate on it on another PR!

oliviertassinari commented:
> see the scrollbar on the demo

There is an open GitHub issue that describes the next step for the scrollbar on dark mode. The plan is to remove the global overrides.

mnajdova commented:
> @mnajdova There is an open GitHub issue that describes the next step for the scrollbar on dark mode. The plan is to remove the global overrides because it comes with too many downsides vs. the alternative.

I've found it, linking it here for future reference https://github.com/mui-org/material-ui/issues/25016 I will revert for the now the changes for the scrollbar, we can come back to this after the v5 stable release.

mnajdova commented:
> These values are just me playing, not necessarily the correct ones. Probably somewhere around 700px - 800px lies a good max-width. For reference, most of our users check the website in a 1920x1080 screen res which renders the div with 1152 width, which is too much 😅 But we can iterate on it on another PR!

Agree, let's keep the scope smaller, so that we can iterate faster :)

mnajdova commented:
> 1. The texts and the demo/code containers are slightly misaligned horizontally. Couldn't find where to play with this...

Fixed with https://github.com/mui-org/material-ui/pull/27860/commits/b2dcfa0cd8e2cda8a9f011e962b0f25a2815a5e8
> 2. There are some cases where the demo and code container don't have border-radius applied. No idea why...

Fixed with https://github.com/mui-org/material-ui/pull/27860/commits/367273c76771b279e2dd98b7382a552ce0297d40

> 3. Couldn't find where to modify the colors of the filled and outlined demo container on dark mode.

We talked offline, we need to find better color over here - https://github.com/mui-org/material-ui/blob/next/docs/src/modules/components/Demo.js#L110

> 4. Couldn't find where to modify the max-width of the container that holds all the content but I played with some values directly on the inspector. This is something I personally think it makes for a much better reading experience (less horizontal movement of the eye) and also for a more overall balanced look since we gain negative space between the sidebar and the table of contents.

You should be able to change it over here - https://github.com/mui-org/material-ui/blob/next/docs/src/modules/components/AppLayoutDocs.js#L36 Not sure what is the best value to be honest.

mnajdova commented:
@danilo-leal leaving a note that we need to find better color for this use-case as well:

<img src="https://user-images.githubusercontent.com/4512430/130567336-c90aef5b-a12c-4173-b6ae-acc569839f25.png" width="400px" />

siriwatknp commented on docs/src/modules/brandingTheme.ts:
```diff
@@ -110,7 +110,7 @@ export const getDesignTokens = (mode: 'light' | 'dark') =>
     palette: {
       primary: blue,
-      divider: mode === 'dark' ? blueDark[400] : grey[200],
+      divider: mode === 'dark' ? blueDark[700] : grey[200],
```
![image](https://user-images.githubusercontent.com/18292247/130598850-95e70ca6-1969-4823-8073-790dc8d090bd.png)

```suggestion
      divider: mode === 'dark' ? blueDark[500] : grey[200],
```
I think it should be lighter than `700`. So far, `700, 800, 900` are background.

siriwatknp commented on docs/src/modules/components/AppTableOfContents.js:
```diff
@@ -13,12 +13,12 @@ const Nav = styled('nav')(({ theme }) => {
     height: 'calc(100vh - 70px)',
     overflowY: 'auto',
-    padding: theme.spacing(2, 2, 2, 0),
+    padding: theme.spacing(2, 5, 2, 0),
```
Are you sure about this? that will reduce the space of text.

siriwatknp reviewed (approved):
Left a couple of comments, the rest looks good.

siriwatknp commented on docs/src/modules/components/MarkdownElement.js:
```diff
@@ -56,32 +55,42 @@ const Root = styled('div')(({ theme }) => ({
   '& h1': {
     ...theme.typography.h3,
     fontSize: 40,
+    fontFamily: ['"PlusJakartaSans"'].join(','),
```
```suggestion
    fontFamily: ['"PlusJakartaSans-ExtraBold"'].join(','),
```

it should have `ExtraBold` as a suffix

mnajdova commented:
> Left a couple of comments, the rest looks good.

I will let @danilo-leal respond. The changes are mostly his :)

mnajdova commented:
I am going to merge this one to leverage some fixes we already have as part of the PR. Will leave the unaddressed comments by @siriwatknp open, so that we can decide and address them in the next PR.

mbrookes commented:
Hmm... the font for `<code>` doesn't align with the body text. It's quite noticeable. We can pick it up in the next iteration.

Also not convinced about the <h1> color - but that can be fixed at any point.
